core/shared/src/main/scala/org/typelevel/log4cats/extras/WriterStructuredLogger.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats.extras
+
+import cats.data.Writer
+import cats.syntax.all._
+import cats.{~>, Alternative, Applicative, Foldable, Id}
+import org.typelevel.log4cats.{SelfAwareStructuredLogger, StructuredLogger}
+
+/**
+ * A `SelfAwareStructuredLogger` implemented using `cats.data.Writer`.
+ *
+ * If a `SelfAwareStructuredLogger` is needed for test code, the `testing` module provides a better
+ * option: `org.typelevel.log4cats.testing.StructuredTestingLogger`
+ */
+object WriterStructuredLogger {
+  def apply[G[_]: Alternative](
+      traceEnabled: Boolean = true,
+      debugEnabled: Boolean = true,
+      infoEnabled: Boolean = true,
+      warnEnabled: Boolean = true,
+      errorEnabled: Boolean = true
+  ): SelfAwareStructuredLogger[Writer[G[StructuredLogMessage], *]] =
+    WriterTStructuredLogger[Id, G](
+      traceEnabled,
+      debugEnabled,
+      infoEnabled,
+      warnEnabled,
+      errorEnabled
+    )
+
+  def run[F[_]: Applicative, G[_]: Foldable](
+      l: StructuredLogger[F]
+  ): Writer[G[StructuredLogMessage], *] ~> F =
+    new ~>[Writer[G[StructuredLogMessage], *], F] {
```

Comment:
Absolutely not a big deal, but I usually write these with infix notation, just like the types.

```scala
new (Writer[G[StructuredLogMessage], *] ~> F) {
```

Reply:
Fair 👍🏻 
This is nearly a direct copy of the existing one, so if you're of the mind to clean this up at some point, it's probably worth changing those to keep things consistent.

Reply:
Yeah, let's just keep these consistent with what we already have.  My way may be idiosyncratic, but it's consistent with how we usually express the return type.

Reply:
I ended up updating this when I went in to update the positive side of the warnings.